### PR TITLE
Readability patch

### PR DESCRIPTION
**Pull request by Lovecrafts-Cat**
This PR aims to increase the readability of the code, while also making it compact through the following commits.
As such, these changes should not be altering any logic/behavior of the code.

This seems to have passed the trial by fire of `.toggle all on`, so I'm quite confident it shouldn't cause any issues.
Feel free to verify it for yourself and report if any arise.

### CODE REVIEW

**Comment by RacoonDog** [comment]
reading through this crashed my browser :))

File: src/main/java/meteordevelopment/meteorclient/gui/GuiThemes.java [hunk 1]
```diff
@@ -111,15 +111,15 @@ public static String[] getNames() {
...
+        if (get() == null) return;
 
-                THEMES_FOLDER.mkdirs();
-                NbtIo.write(tag, new File(THEMES_FOLDER, get().name + ".nbt"));
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
+        try {
+            NbtCompound tag = get().toTag();
+
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/GuiThemes.java:118]: remove this

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java [hunk 2]
```diff
@@ -45,12 +45,14 @@ public abstract class WidgetScreen extends Screen {
...
 
-    public boolean locked, lockedAllowClose;
+    public boolean locked;
+    public boolean lockedAllowClose;
     private boolean closed;
     private boolean onClose;
     private boolean debug;
 
-    private double lastMouseX, lastMouseY;
+    private double lastMouseX;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java:54]: why

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java [hunk 3]
```diff
@@ -45,12 +45,14 @@ public abstract class WidgetScreen extends Screen {
 
     protected final GuiTheme theme;
 
-    public boolean locked, lockedAllowClose;
+    public boolean locked;
+    public boolean lockedAllowClose;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java:49]: why

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java [hunk 4]
```diff
@@ -70,9 +72,7 @@ public WidgetScreen(GuiTheme theme, String title) {
         if (parent != null) {
             animProgress = 1;
 
-            if (this instanceof TabScreen && parent instanceof TabScreen) {
-                parent = ((TabScreen) parent).parent;
-            }
+            if (this instanceof TabScreen && parent instanceof TabScreen) parent = ((TabScreen) parent).parent;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java:75]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java [hunk 5]
```diff
@@ -331,20 +321,18 @@ public void removed() {
                 for (Runnable action : onClosed) action.run();
             }
 
-            if (onClose) {
-                taskAfterRender = () -> {
-                    locked = true;
-                    mc.setScreen(parent);
-                };
-            }
+            if (onClose) taskAfterRender = () -> {
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java:324]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java [hunk 6]
```diff
@@ -318,10 +311,7 @@ public void removed() {
             Input.setCursorStyle(CursorStyle.Default);
 
             loopWidgets(root, widget -> {
-                if (widget instanceof WTextBox textBox) {
-
-                    if (textBox.isFocused()) textBox.setFocused(false);
-                }
+                if (widget instanceof WTextBox textBox && textBox.isFocused()) textBox.setFocused(false);
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java:314]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/AddAccountScreen.java [hunk 7]
```diff
@@ -25,15 +25,8 @@ public void tick() {
...
-            }
-            else {
-                timer++;
-            }
-        }
-
-        else if (!add.getText().equals("Add")) {
-            add.set("Add");
-        }
+            } else timer++;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/AddAccountScreen.java:28]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/AddAccountScreen.java [hunk 8]
```diff
@@ -25,15 +25,8 @@ public void tick() {
...
+        } else if (!add.getText().equals("Add")) add.set("Add");
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/AddAccountScreen.java:29]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/AddAlteningAccountScreen.java [hunk 9]
```diff
@@ -28,9 +28,7 @@ public void initWidgets() {
         // Add
         add = t.add(theme.button("Add")).expandX().widget();
         add.action = () -> {
-            if (!token.get().isEmpty()) {
-                AccountsScreen.addAccount(this, parent, new TheAlteningAccount(token.get()));
-            }
+            if (!token.get().isEmpty()) AccountsScreen.addAccount(this, parent, new TheAlteningAccount(token.get()));
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/AddAlteningAccountScreen.java:31]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/ModulesScreen.java [hunk 10]
```diff
@@ -59,9 +59,7 @@ protected WWindow createCategory(WContainer c, Category category) {
         w.padding = 0;
         w.spacing = 0;
 
-        if (theme.categoryIcons()) {
-            w.beforeHeaderInit = wContainer -> wContainer.add(theme.item(category.icon)).pad(2);
-        }
+        if (theme.categoryIcons()) w.beforeHeaderInit = wContainer -> wContainer.add(theme.item(category.icon)).pad(2);
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/ModulesScreen.java:62]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/AddCrackedAccountScreen.java [hunk 11]
```diff
@@ -34,9 +34,7 @@ public void initWidgets() {
         add.action = () -> {
             if (!name.get().isEmpty() && (name.get().length() < 17) && name.get().matches("^[a-zA-Z0-9_]+$")) {
                 CrackedAccount account = new CrackedAccount(name.get());
-                if (!(Accounts.get().exists(account))) {
-                    AccountsScreen.addAccount(this, parent, account);
-                }
+                if (!Accounts.get().exists(account)) AccountsScreen.addAccount(this, parent, account);
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/AddCrackedAccountScreen.java:37]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/NotebotSongsScreen.java [hunk 12]
```diff
@@ -73,9 +73,7 @@ private void initSongsTable() {
             table.row();
         }
 
-        if (noSongsFound.get()) {
-            table.add(theme.label("No songs found.")).expandCellX().center();
-        }
+        if (noSongsFound.get()) table.add(theme.label("No songs found.")).expandCellX().center();
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/NotebotSongsScreen.java:76]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java [hunk 13]
```diff
@@ -40,7 +38,8 @@ public void initWidgets() {
             WVerticalList list = add(theme.section("Log", false)).widget().add(theme.verticalList()).expandX().widget();
             Proxies proxies = Proxies.get();
             try {
-                int pog = 0, bruh = 0;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java:43]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java [hunk 14]
```diff
@@ -57,28 +56,26 @@ public void initWidgets() {
...
+                    .label("Successfully imported " + success + "/" + (failure + success) + " proxies.")
+                    .color(Utils.lerp(Color.RED, Color.GREEN, (float) success / (success + failure)))
                 );
             } catch (IOException e) {
                 e.printStackTrace();
             }
-        } else {
-            add(theme.label("Invalid File!"));
-        }
+        } else add(theme.label("Invalid File!"));
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java:78]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/settings/ColorSettingScreen.java [hunk 15]
```diff
@@ -461,9 +453,7 @@ void calculateFromSetting(boolean calculateNow) {
             if (calculateNow) {
                 double huePercentage = hueAngle / 360;
                 handleX = huePercentage * width;
-            } else {
-                calculateHandleXOnLayout = true;
-            }
+            } else calculateHandleXOnLayout = true;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/settings/ColorSettingScreen.java:456]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/settings/ColorSettingScreen.java [hunk 16]
```diff
@@ -566,10 +554,8 @@ public void onMouseMoved(double mouseX, double mouseY, double lastMouseX, double
                 if (mouseX >= this.x && mouseX <= this.x + width) {
                     handleX += mouseX - lastMouseX;
                     handleX = Utils.clamp(handleX, 0, width);
-                } else {
-                    if (handleX > 0 && mouseX < this.x) handleX = 0;
-                    else if (handleX < width && mouseX > this.x + width) handleX = width;
-                }
+                } else if (handleX > 0 && mouseX < this.x) handleX = 0;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/settings/ColorSettingScreen.java:557]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java [hunk 17]
```diff
@@ -69,14 +69,12 @@ public void initWidgets() {
...
 
-            if (!setting.onlyAttackable || EntityUtils.isAttackable(entityType)) {
-                switch (entityType.getSpawnGroup()) {
-                    case CREATURE -> hasAnimal++;
-                    case WATER_AMBIENT, WATER_CREATURE, UNDERGROUND_WATER_CREATURE, AXOLOTLS -> hasWaterAnimal++;
-                    case MONSTER -> hasMonster++;
-                    case AMBIENT -> hasAmbient++;
-                    case MISC -> hasMisc++;
-                }
+            if (!setting.onlyAttackable || EntityUtils.isAttackable(entityType)) switch (entityType.getSpawnGroup()) {
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java:72]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java [hunk 18]
```diff
@@ -133,36 +131,33 @@ public void initWidgets() {
...
-                    }
-                    case AMBIENT -> {
-                        ambientE.add(entityType);
-                        addEntityType(ambientT, ambientC, entityType);
-                    }
-                    case MISC -> {
-                        miscE.add(entityType);
-                        addEntityType(miscT, miscC, entityType);
-                    }
+            if (!setting.onlyAttackable || EntityUtils.isAttackable(entityType)) switch (entityType.getSpawnGroup()) {
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java:134]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java [hunk 19]
```diff
@@ -207,11 +202,7 @@ private void tableChecked(List<EntityType<?>> entityTypes, boolean checked) {
             if (checked) {
                 setting.get().put(entityType, true);
                 changed = true;
-            } else {
-                if (setting.get().removeBoolean(entityType)) {
-                    changed = true;
-                }
-            }
+            } else if (setting.get().removeBoolean(entityType)) changed = true;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java:205]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java [hunk 20]
```diff
@@ -250,30 +241,26 @@ private void addEntityType(WTable table, WCheckbox tableCheckbox, EntityType<?>
...
-                        case AMBIENT -> {
-                            hasAmbient--;
-                            if (hasAmbient == 0) tableCheckbox.checked = false;
-                        }
-                        case MISC -> {
-                            hasMisc--;
-                            if (hasMisc == 0) tableCheckbox.checked = false;
-                        }
-                    }
+            } else if (setting.get().removeBoolean(entityType)) switch (entityType.getSpawnGroup()) {
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/settings/EntityTypeListSettingScreen.java:244]: 👎👎👎👎👎

File: src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/BaritoneTab.java [hunk 21]
```diff
@@ -57,61 +57,55 @@ private static Settings getSettings() {
...
-                    sgString.add(new StringSetting.Builder()
-                        .name(setting.getName())
-                        .description(getDescription(setting.getName()))
-                        .defaultValue((String) setting.defaultValue)
-                        .onChanged(string -> setting.value = string)
-                        .onModuleActivated(stringSetting -> stringSetting.set((String) setting.value))
-                        .build()
-                    );
-                } else if (value instanceof Color) {
+                if (value instanceof Boolean) sgBool.add(new BoolSetting.Builder()
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/BaritoneTab.java:60]: 👎👎👎👎👎👎👎👎

File: src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/BaritoneTab.java [hunk 22]
```diff
@@ -126,25 +120,22 @@ private static Settings getSettings() {
...
-                        sgItemLists.add(new ItemListSetting.Builder()
-                            .name(setting.getName())
-                            .description(getDescription(setting.getName()))
-                            .defaultValue((List<Item>) setting.defaultValue)
-                            .onChanged(itemList -> setting.value = itemList)
-                            .onModuleActivated(itemListSetting -> itemListSetting.set((List<Item>) setting.value))
-                            .build()
-                        );
-                    }
+                    if (type == Block.class) sgBlockLists.add(new BlockListSetting.Builder()
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/BaritoneTab.java:123]: 👎👎👎👎

File: src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ConfigTab.java [hunk 23]
```diff
@@ -42,36 +42,30 @@ public ConfigScreen(GuiTheme theme, Tab tab) {
...
-                    YesNoPrompt.create(theme, this.parent)
-                        .title("Long command prefix")
-                        .message("You have set your command prefix to a very long string.")
-                        .message("This means that in order to execute any command, you will need to type %s followed by the command you want to run.", prefix)
-                        .message("Do you want to reset your prefix back to '.'?")
-                        .onYes(() -> Config.get().prefix.set("."))
-                        .id("long-command-prefix")
-                        .show();
-                }
+                if (prefix.isBlank()) YesNoPrompt.create(theme, this.parent)
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ConfigTab.java:45]: 👎👎👎👎👎👎

File: src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/MeteorGuiTheme.java [hunk 24]
```diff
@@ -54,7 +54,7 @@ public class MeteorGuiTheme extends GuiTheme {
             .sliderRange(0.75, 4)
             .onSliderRelease()
             .onChanged(aDouble -> {
-                if (mc.currentScreen instanceof WidgetScreen) ((WidgetScreen) mc.currentScreen).invalidate();
+                if (mc.currentScreen instanceof WidgetScreen widgetScreen) widgetScreen.invalidate();
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/MeteorGuiTheme.java:57]: braces

File: src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/MeteorGuiTheme.java [hunk 25]
```diff
@@ -371,7 +371,9 @@ public boolean hideHUD() {
     }
 
     public class ThreeStateColorSetting {
-        private final Setting<SettingColor> normal, hovered, pressed;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/MeteorGuiTheme.java:374]: 👎👎

File: src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/widgets/input/WMeteorTextBox.java [hunk 26]
```diff
@@ -100,10 +100,7 @@ protected void onRender(GuiRenderer renderer, double mouseX, double mouseY, doub
         if (cursorTimer >= 1) {
             cursorVisible = !cursorVisible;
             cursorTimer = 0;
-        }
-        else {
-            cursorTimer += delta * 1.75;
-        }
+        } else cursorTimer += delta * 1.75;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/widgets/input/WMeteorTextBox.java:103]: 👎👎👎👎

File: src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/widgets/pressable/WMeteorButton.java [hunk 27]
```diff
@@ -23,9 +23,7 @@ protected void onRender(GuiRenderer renderer, double mouseX, double mouseY, doub
 
         renderBackground(renderer, this, pressed, mouseOver);
 
-        if (text != null) {
-            renderer.text(text, x + width / 2 - textWidth / 2, y + pad, theme.textColor.get(), false);
-        }
+        if (text != null) renderer.text(text, x + width / 2 - textWidth / 2, y + pad, theme.textColor.get(), false);
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/widgets/pressable/WMeteorButton.java:26]: 👎

File: src/main/java/meteordevelopment/meteorclient/gui/utils/Cell.java [hunk 28]
```diff
@@ -158,23 +158,19 @@ public void alignWidget() {
         if (expandWidgetX) {
             widget.x = x;
             widget.width = width;
-        } else {
-            switch (alignX) {
-                case Left -> widget.x = x;
-                case Center -> widget.x = x + width / 2 - widget.width / 2;
-                case Right -> widget.x = x + width - widget.width;
-            }
+        } else switch (alignX) {
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/gui/utils/Cell.java:161]: giving up

**Comment by RickyTheRacc** [comment]
Honestly quite incredible

File: src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java (hunk 4 above)
Comment (RacoonDog) [reply src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java]: imo this one is fine if you use pattern matching to remove the cast

Comment (RacoonDog) [reply src/main/java/meteordevelopment/meteorclient/gui/WidgetScreen.java]: e.g. `if (this instanceof TabScreen && parent instanceof TabScreen parentTabScreen) parent = parentTabScreen.parent;`

File: src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java (hunk 13 above)
Comment (RacoonDog) [reply src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java:43]: bruh

File: src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/MeteorGuiTheme.java (hunk 24 above)
Comment (RacoonDog) [reply src/main/java/meteordevelopment/meteorclient/gui/themes/meteor/MeteorGuiTheme.java]: this one is yes

**Comment by Lovecrafts-Cat** [comment]
Going to close this PR, as it doens't really do anything good for the project itself.
Too many changes for so little gain; only notable improvement would have been the `instanceof` pattern matching.

**Pull request closed without merging**